wa/workloads/antutu/__init__.py

```diff
@@ -109,16 +146,109 @@ def extract_scores(self, context, regex_version):
                             result = float('NaN')
                         entry = regex.pattern.rsplit(None, 1)[0]
                         context.add_metric(entry, result, lower_is_better=False)
+                        #Calculate group scores if 'CPU' in entry:
+                        if 'CPU' in entry:
+                            cpu.append(result)
+                            cpu_result = sum(cpu)
+                        if 'GPU' in entry:
+                            gpu.append(result)
+                            gpu_result = sum(gpu)
+                        if any([i in entry for i in ['Data', 'Document', 'Image', 'User', 'Video']]):
+                            ux.append(result)
+                            ux_result = sum(ux)
+                        if any([i in entry for i in ['RAM', 'ROM']]):
+                            mem.append(result)
+                            mem_result = sum(mem)
                         expected_results -= 1
         if expected_results > 0:
             msg = "The Antutu workload has failed. Expected {} scores, Detected {} scores."
             raise WorkloadError(msg.format(len(regex_version), expected_results))
 
+        context.add_metric('CPU Total Score', cpu_result, lower_is_better=False)
+        context.add_metric('GPU Total Score', gpu_result, lower_is_better=False)
+        context.add_metric('UX Total Score', ux_result, lower_is_better=False)
+        context.add_metric('MEM Total Score', mem_result, lower_is_better=False)
+
+        #Calculate overall scores
+        overall_result = float(cpu_result + gpu_result + ux_result + mem_result)
+        context.add_metric('Overall Score', overall_result, lower_is_better=False)
+
     def update_output(self, context):
         super(Antutu, self).update_output(context)
+        if self.version.startswith('10'):
+            self.extract_scores(context, self.regex_matches_v10)
         if self.version.startswith('9'):
             self.extract_scores(context, self.regex_matches_v9)
         if self.version.startswith('8'):
             self.extract_scores(context, self.regex_matches_v8)
         if self.version.startswith('7'):
             self.extract_scores(context, self.regex_matches_v7)
+
+
+class AntutuBDP(Workload):
```

Comment:
High level question, what was the reason that we're using the generic `Workload` as opposed to the `ApkWorkload` class which could have handled some of the generic APK management automatically?

Reply:
I went with the Workload class as although it's installing an APK it is executing a command line argument. Is that an incorrect approach? Would you prefer this to be an ApkWorkload?